server/src/main/scala/org/ensime/model/ModelBuilders.scala

```diff
@@ -39,29 +39,33 @@ trait ModelBuilders { self: RichPresentationCompiler =>
     }
   }
 
-  def locateSymbolPos(sym: Symbol, needPos: PosNeeded): Option[SourcePosition] =
-    if (sym == NoSymbol || needPos == PosNeededNo) None
+  def locateSymbolPos(sym: Symbol, needPos: PosNeeded): Option[SourcePosition] = {
+    if (sym == NoSymbol || needPos == PosNeededNo)
+      None
     else if (sym.pos != NoPosition) {
-      if (needPos eq PosNeededYes)
+      if (needPos == PosNeededYes || needPos == PosNeededAvail) {
```

Comment:
Would you mind doing a performance check for this? Sending all known symbol positions can produce huge messages. Good examples are inspecting  the RichPresentationCompiler class or the org.ensime package.